frontend/src/composables/formAutoSave.ts

```diff
@@ -0,0 +1,46 @@
+import { ref, onMounted, onBeforeUnmount } from 'vue';
+
+// autosave functionality for forms, saves on inactivity after the delay (default 10 seconds)
+export function useAutoSave(saveFunc: () => Promise<void>, delay: number = 10000) {
+  const formUpdated = ref(false);
```

Comment:
Variables should declare their type if possible. `const formUpdated: Ref<boolean> = ref(false);`